app/assets/javascripts/institution-ror-typeahead.js

```diff
@@ -0,0 +1,266 @@
+var ROR_API_URL = "https://api.ror.org/organizations";
+
+
+function toggleUserInput(disabled) {
+    const action = disabled ? 'addClass' : 'removeClass';
+    const elements = [
+        '#institution_title',
+        '#institution_city',
+        '#institution_ror_id',
+        '#institution_web_page',
+        '.tt-input'
+    ];
+
+    elements.forEach(selector => {
+        $j(selector)[action]('institution-input-disable');
+        $j(selector).prop("readonly", disabled);
+        $j('#institution_country').prop('disabled', disabled);
+    });
+}
+
+function extractRorId(rorUrl) {
+    const regex = /https:\/\/ror\.org\/([^\/]+)/;
+    const match = rorUrl.match(regex);
+    if (match) {
+        return match[1];
+    } else {
+        return null;
+    }
+}
+
+function fetchRorData(rorId) {
+    var url = ROR_API_URL + '/' + rorId;
```

Comment:
is it safe to do this, without explicity asking a users consent? Besides possible cross-origin issues, doesn't this introduce the same issue of passing the IP address to another server, similar to what led to preventing us using google fonts, and embedded youtube videos ?

https://www.cookieyes.com/documentation/google-fonts-and-gdpr/